### PR TITLE
Feature/55 setup automatic release

### LINKED ISSUE #55: Setup automatic release
---

PR # (fill in once PR number is available)

---

**About**
With the version `1.1.0` of shared library, new utility functions for automatic releases can be implemented.

---

- [ ] Update `adverity-shared-library`
- [ ] setup jenkins job

### PR DESCRIPTION
**Pull request by reakaleek**
Closes #55

### CODE REVIEW

**Comment by reakaleek** [comment]
run pr-checks

File: README.md [hunk 1]
```diff
@@ -28,3 +28,8 @@ Use it by setting the extends field in your `.eslintrc`:
   extends: '@pmedianetwork/eslint-config/typescript',
 };
 ```
+
+
+## Release Management
+
+See http://handbook.adverity.net/changesets/#the-library-changesets-workflow
```
Comment (reakaleek) [inline README.md]: Will be available when pmedianetwork/handbook#50 is merged

**Comment by reakaleek** [comment]
Ignore `pr-checks` this was removed/renamed to `changeset-validation`

**Review by danielbartsch — APPROVED** [review]

File: .jenkins/changeset-validation.Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    libraries { lib 'adverity-shared-library@2.1.0' }
+    agent { label "PRJob" }
+    environment {
+        NODE_VERSION = getNodeVersion()
+        GITHUB_TOKEN = credentials('e8c4eee6-cef3-4fd5-a65c-1050f7ecb0c7')
+    }
+    stages {
+        stage('Validate changesets') {
+            steps {
+                validateChangesets()
```
Comment (TkDodo) [inline .jenkins/changeset-validation.Jenkinsfile]: can we add the validation that every changeset needs an `issue` linked ?

**Review by TkDodo — CHANGES_REQUESTED** [review]
Also: if we want to deploy multiple things with one version, how would we do that? Would we need to create some other branch to merge to instead of `develop` ?

**Comment by reakaleek** [comment]
> Also: if we want to deploy multiple things with one version, how would we do that? Would we need to create some other branch to merge to instead of `develop` ?

http://handbook.adverity.net/changesets/#the-library-changesets-workflow
> If you want to release multiple features or fixes in a single release, you have to create an epic branch.

File: .jenkins/changeset-validation.Jenkinsfile (hunk 2 above)
Comment (reakaleek) [reply .jenkins/changeset-validation.Jenkinsfile]: https://github.com/pmedianetwork/eslint-config/pull/56/commits/35031624d9b1565008e8bab2f32b410bd75a85f9

**Review by TkDodo — APPROVED** [review]

**Pull request merged**